dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/util/ContentletUtilTest.java

```diff
@@ -377,43 +273,34 @@ public void test_getContentPrintableMap_WhenContentTypeIsFileAsset_PathIsAddedTo
         }
     }
 
-    private ContentType createContentType(final BaseContentType baseContentType)
-            throws DotSecurityException, DotDataException {
-
-        final long i = System.currentTimeMillis();
-        //Create Content Type.
-        final ContentType contentType = ContentTypeBuilder.builder(baseContentType.immutableClass())
-                .description("Test ContentType" + i)
-                .host(defaultHost.getIdentifier())
-                .name("Test ContentType" + i)
-                .owner("owner")
-                .variable("testVelocityVarName")
-                .build();
-
-        return contentTypeAPI.save(contentType);
+    private CategoryDataGen createCategory(final String categoryName, final int sortOrder) {
+        final SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyyMMdd-HHmmss");
+        final String categoryKey = categoryName.toLowerCase().replaceAll("\\s", "")
+                + "-" + simpleDateFormat.format(new Date());
+        return new CategoryDataGen().setCategoryName(categoryName)
+            .setKey(categoryKey).setCategoryVelocityVarName(categoryKey)
+            .setSortOrder(sortOrder);
     }
 
-
-    /**
-     * Util method to check if NotFoundInDbException is returned when trying to find a Field or ContentType.
-     * @param o ContentType or Field to find.
-     * @return true is NotFoundInDbException is returned, false if else.
-     */
-    private boolean checkNotFoundInDbException (Object o){
-        if ( o instanceof ContentType ){
-            try {
-                contentTypeAPI.find( ((ContentType)o).inode() );
-            } catch ( Exception e ){
-                return true;
-            }
-        }
-        if ( o instanceof Field ){
-            try {
-                fieldAPI.find( ((Field)o).id() );
-            } catch ( Exception e ){
-                return true;
-            }
+    private void verifyCategoriesForField(
+            final Map<String, Object> contentPrintableMap,
+            final String fieldName, final Category ...categories) {
+
+        final List<?> categoryList = (List<?>) contentPrintableMap.get(fieldName);
+        assertNotNull(categoryList);
+        assertEquals(categories.length, categoryList.size());
+        for (final Category category : categories) {
+            Optional<?> matchElement = categoryList.stream()
+                    .filter(map -> {
+                        final Map<?,?> cMap = (Map<?,?>) map;
+                        return cMap.containsKey("key")
+                                && cMap.get("key").equals(category.getKey());
+                    }).findFirst();
+            assertTrue(matchElement.isPresent());
+            Map<?,?> categoryMap = (Map<?, ?>) matchElement.get();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'categoryMap' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4885353)